zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -1361,27 +1386,39 @@ private Record processSasl(ByteBuffer incomingBuffer, ServerCnxn cnxn) throws IO
                         cnxn.addAuthInfo(new Id("super", ""));
                     }
                 }
-            }
-            catch (SaslException e) {
+            } catch (SaslException e) {
                 LOG.warn("Client failed to SASL authenticate: " + e, e);
-                if ((System.getProperty("zookeeper.allowSaslFailedClients") != null)
-                  &&
-                  (System.getProperty("zookeeper.allowSaslFailedClients").equals("true"))) {
-                    LOG.warn("Maintaining client connection despite SASL authentication failure.");
+                if (shouldAllowSaslFailedClientsConnect() && !shouldRequireClientSaslAuth()) {
+                  LOG.warn("Maintaining client connection despite SASL authentication failure.");
                 } else {
+                  int error;
+                  if (shouldRequireClientSaslAuth()) {
+                    LOG.warn("Closing client connection due to server requires client SASL authenticaiton," +
```

Comment:
Is it possible to report at least the client ip address? This way it is simpler to correlate the errors.
Not blocker for me, this class already has logs without references to the client. It could be a more global enhancement for the future